lib/workers/repository/update/pr/automerge.ts

```diff
@@ -69,6 +69,10 @@ export async function checkAutoMerge(
       prAutomergeBlockReason: 'PlatformNotReady',
     };
   }
+  if (config.autoApprove && platform.approvePrForAutomerge) {
+    logger.debug('Auto-approving PR for automerge');
+    await platform.approvePrForAutomerge(pr.number);
```

Comment:
it this a noop if it's already approved? see other open conversation

Reply:
But the PR could have been un-approved after initial approve. See the PR body's _Context_:

> I self-host Renovate and I noticed that auto-merge was not working as expected. I realized it was caused by this issue, because for a given project we have here, the same user/robot also votes the change with a Code-Review +1 vote when SonarQube analysis is successful (which is a very known pattern).

SonarQube analysis in such case happens during CI, after Renovate has initially approved the PR upon creation or update.

This extra-check will account for cases where the PR had its vote downgraded from +2 to +1.

Reply:
I believe this is a special case that can only happen with Gerrit. On GitHub and Gitlab, there's no such concept of approval level. It's either approved or not, by a given user.

However, in Gerrit, "approvals" can either be +2 (highest, needed to merge), or +1 (not enough to merge). And a user, can have its vote decreased from +2 to +1.

Reply:
@viceice, I believe this was the last concern on this PR.

Have you had a chance to re-think it?

I can probably break this PR into two PRs by the way. One for the saving API call improvement, and another one fixing the issue reported.